packages/now-build-utils/src/fs/node-version.ts

```diff
@@ -26,77 +32,83 @@ export function getLatestNodeVersion(): NodeVersion {
-            : '')
-      );
+    if (!found) {
+      const intro = isAuto
+        ? 'This project is using an invalid version of Node.js and must be changed.'
+        : 'Found `engines` in `package.json` with an invalid Node.js version range: ' +
+          engineRange;
+      throw new NowBuildError({
+        code: 'NOW_BUILD_UTILS_NODE_VERSION_INVALID',
+        message:
+          intro +
+          '\n' +
+          pleaseUse +
+          JSON.stringify(supportedOptions.map(o => o.range)),
+      });
     }
   }
 
-  const { range, discontinueDate } = selection;
-  if (discontinueDate && !isDiscontinued(selection)) {
-    const d = discontinueDate.toISOString().split('T')[0];
+  if (isDiscontinued(selection)) {
+    const intro = isAuto
+      ? 'This project is using a discontinued version of Node.js and must be upgraded.'
+      : 'Found `engines` in `package.json` with a discontinued Node.js version range: ' +
+        engineRange;
+    throw new NowBuildError({
+      code: 'NOW_BUILD_UTILS_NODE_VERSION_DISCONTINUED',
+      message:
+        intro +
+        '\n' +
+        pleaseUse +
+        JSON.stringify(supportedOptions.map(o => o.range)) +
+        '\n' +
+        upstreamProvider,
+    });
+  }
+
+  debug(
+    isAuto
+      ? 'Using default Node.js range: ' + selection.range
+      : (engineRange ? 'Found' : 'Missing') +
+          ' `engines` in `package.json`, selecting range: ' +
+          selection.range
+  );
+
+  if (selection.discontinueDate) {
+    const d = selection.discontinueDate.toISOString().split('T')[0];
     const validRanges = supportedOptions
       .filter(o => !o.discontinueDate)
       .map(o => o.range);
-    const prevTerm = process.env.TERM;
-    if (!prevTerm) {
-      // workaround for https://github.com/sindresorhus/term-size/issues/13
-      process.env.TERM = 'xterm';
-    }
     console.warn(
       boxen(
         'NOTICE' +
           '\n' +
-          `\nNode.js version ${range} has reached end-of-life.` +
+          `\nNode.js version ${selection.range} has reached end-of-life.` +
           `\nAs a result, deployments created on or after ${d} will fail to build.` +
-          '\nPlease use one of the following supported `engines` in `package.json`: ' +
+          '\n' +
+          pleaseUse +
           JSON.stringify(validRanges) +
-          '\nThis change is the result of a decision made by an upstream infrastructure provider (AWS).' +
-          '\nRead more: https://docs.aws.amazon.com/lambda/latest/dg/runtime-support-policy.html',
+          '\n' +
+          upstreamProvider,
         { padding: 1 }
       )
     );
-    process.env.TERM = prevTerm;
   }
 
   return selection;
 }
 
 function isDiscontinued({ discontinueDate }: NodeVersion): boolean {
-  const today = new Date();
-  return discontinueDate !== undefined && discontinueDate <= today;
+  const today = Date.now();
+  return discontinueDate !== undefined && discontinueDate.getTime() <= today;
```

Comment:
Up to you, but the`getTime()` is not necessary here. `Date#valueOf()` already returns the millisecond value so `<=` coerces the Date instance.

```suggestion
  return discontinueDate !== undefined && discontinueDate <= today;
```

Reply:
Good point. I think I'll leave it instead of coercing because it makes more sense in my brain with two numbers as operands instead of mixed types.